test/jdk/java/awt/dnd/NextDropActionTest/NextDropActionTest.java

```diff
@@ -0,0 +1,203 @@
+                public void dragDropEnd(DragSourceDropEvent e) {
+                    System.err.println("DragSourseListener.dragDropEnd(): " +
+                            " firstCall=" + firstCall +
+                            " drop action=" + e.getDropAction());
+                    if (firstCall) {
+                        firstCall = false;
+                        synchronized (LOCK) {
+                            firstEnd = true;
+                            LOCK.notifyAll();
+                        }
+                        return;
+                    }
+                    if (e.getDropAction() != DnDConstants.ACTION_MOVE) {
+                        System.err.println("FAILURE: wrong drop action:"
+                                + e.getDropAction());
+                        failed = true;
+                    }
+                    synchronized (LOCK) {
+                        secondEnd = true;
+                        LOCK.notifyAll();
+                    }
+                }
+            };
+
+            DragGestureListener dgl = dge ->
+                    dge.startDrag(null, new StringSelection("test"), dsl);
+
+            new DragSource().createDefaultDragGestureRecognizer(panel,
+                    DnDConstants.ACTION_COPY_OR_MOVE, dgl);
+
+            DropTargetListener dtl = new DropTargetAdapter() {
+                public void drop(DropTargetDropEvent e) {
+                    System.err.println("DropTargetListener.drop(): " +
+                            "accepting the user drop action=" + e.getDropAction());
+                    e.acceptDrop(e.getDropAction());
+                    e.dropComplete(true);
+                }
+            };
+
+            new DropTarget(frame, DnDConstants.ACTION_COPY_OR_MOVE, dtl);
+
+            Point startPoint = new Point(Util.blockTillDisplayed(panel));
+            startPoint.translate(50, 50);
+            Point endPoint = new Point(startPoint.x
+                    + DragSource.getDragThreshold() + 10,
+                    startPoint.y + DragSource.getDragThreshold() + 10);
+
+            synchronized (LOCK) {
+                robot.keyPress(KeyEvent.VK_CONTROL);
+                Util.doDragDrop(robot, startPoint, endPoint);
+                robot.keyRelease(KeyEvent.VK_CONTROL);
+                LOCK.wait(WAIT_TIMEOUT);
+            }
+            if (!firstEnd) {
+                System.err.println("DragSourseListener.dragDropEnd() " +
+                        "was not called, returning");
+                return;
+            }
+
+            synchronized (LOCK) {
+                Util.doDragDrop(robot, startPoint, endPoint);
+                LOCK.wait(WAIT_TIMEOUT);
+            }
+            if (!secondEnd) {
+                System.err.println("DragSourseListener.dragDropEnd() " +
+                        "was not called, returning");
+                return;
+            }
+        } finally {
+            if (frame != null) {
```

Comment:
frame is created with in EDT but disposing without EDT ?

Reply:
There's no harm in doing that but I'm not sure its necessary for the dispose().

Reply:
it is fine to call dispose here on the main, but the frame possibly can be null here.

Reply:
I have updated the tests to all dispose on the EDT just in case.

Reply:
The problem is not in the "dispose" but in the next check "frame != null" the frame variable can be null here even if frame is shown on the screen.